core/include/tangram/map.h

```diff
@@ -129,6 +139,8 @@ class Map {
     // The callback may be be called from the main or worker thread.
     void setSceneReadyListener(SceneReadyCallback _onSceneReady);
 
+    void setCameraAnimationListener(CameraAnimationCallback _cb);
```

Comment:
@hjanetzek what do you think about taking a camera callback as an argument to each animation call? Like
```
void setCameraPositionEased(const CameraPosition& _camera, float duration, EaseType _e, CameraAnimationCallback _cb);
```
The way I see callbacks being used is to trigger events in an app UI after the camera has finished moving to a specific location. This behavior is probably different for each call site, so rather than having one callback set for the map and switching behavior based on state inside the callback, we can make the state implicit by associating one callback with each animation call.

Reply:
Yes we could also do it that way. I made it analog to the `SceneReadyCallback`, though on the Android side the event already is dispatched to the callback passed to `setCameraPositionEased`. 
```
 public void setCameraPositionEased(@NonNull final CameraPosition camera, final int duration, @NonNull final EaseType ease, final CameraAnimationCallback cb) 
```

Reply:
At least for Android it would not help much I guess. Holding onto the java callback object through the native callback function seems more complicated.

Reply:
Hello, sorry for the interruption, but I needed some parity in Tangram-ES with Google Maps on Android so I have implemented cancel and finish callbacks into all easing animations - position, zoom, rotation and tilt. I have working implementation also in production app, but it is in private repository on GitLab, but I can commit these changes into my Tangram-ES forked GitHub repository and maybe try to merge it into this repo? 

I haven't done so before because I am not sure if you wanted to add this functionality and I am not sure if I have done it right way. And also this is rather big change to the interface. Merging my changes into this repository might be also for benefit for me because I will have less conflicts when pulling your changes into my repo.

Reply:
@rwrx Can you describe what animation features you specifically needed for your application? We weren't planning on having separate animation calls for each camera parameter but maybe there are use cases for these that I haven't foreseen.

Reply:
@matteblair I just needed api which can do position, zoom, rotation and tilt animations, because I am also using Google Maps API in my app and I needed also cancel and finish callbacks, mainly due to some logic inside app - for example during the animation I needed to disable some functionality and then right after the animation finished I needed to enable that functionality again. Here is this API in Google Maps: https://developers.google.com/android/reference/com/google/android/gms/maps/GoogleMap#animateCamera(com.google.android.gms.maps.CameraUpdate,%20com.google.android.gms.maps.GoogleMap.CancelableCallback).

I am not sure what you meant by "separate animation calls for each camera parameter", because Tangram-ES already have Map::setPositionEased(), Map::setZoomEased(), Map::setRotationEased and Map::setTiltEased.

Reply:
> At least for Android it would not help much I guess. Holding onto the java callback object through the native callback function seems more complicated.

This was also the reason for not doing it for `SceneReadyCallback`. But I think we did discuss the complexity of doing it to not be huge!

Reply:
@rwrx I see, the aim of this work is to be as close to what google and other mapping provider apis are with camera updates. Can you provide a link to your work (maybe gitlab link itself)?

Reply:
@tallytalwar I have tried to merge my GitLab repo into GitHub fork of Tangram ES, I am not completely sure if it was merged correctly (I haven't tried to compile it), but for example you can see how I have done this easing here: https://github.com/rwrx/tangram-es/blob/master/core/src/map.cpp#L612. Also support for these callbacks has changes in other files, but currently I can't remember exact parts which I have changed as I have done this one year ago. And also it is possible that these parts are merged incorrectly. I just wanted to give you some overview.